nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -117,6 +115,22 @@ in
       '';
     };
 
+    networking.dhcpcd.setHostname = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = ''
+          Whenever to set the machine hostname based on the information
```

Comment:
I think you mean "Whether" instead of "Whenever" here.

Reply:
Fixed